python/BioSimSpace/FreeEnergy/_relative.py

```diff
@@ -696,6 +1278,159 @@ def _difference(self, pmf_ref):
         # Now call the staticmethod passing in both PMFs.
         return Relative.difference(pmf, pmf_ref)
 
+    @staticmethod
+    def check_overlap(overlap, estimator="MBAR"):
+        """Check the overlap of an FEP leg. 
+
+           Parameters
+           ----------
+
+           overlap : [ [ float, float, ... ] ], numpy.matrix
+               The overlap matrix. This gives the overlap between lambda windows.
+
+           estimator : str
+               Must be "MBAR" for checking the overlap matrix.
+
+           Returns
+           -------
+
+           Message - gives a warning if the overlap is less than 0.03 for any off-diagonal.
+
+        """
+        if not isinstance(overlap, _np.matrix):
+            raise TypeError("'overlap' must be of type 'numpy.matrix'.")
+
+        # estimator must be MBAR for overlap matrix or TI for dhdl plot.
+        if estimator not in ['MBAR']:
+            raise ValueError("'estimator' must be 'MBAR'.")
+
+        if estimator == "MBAR":
+            # check the overlap
+            # get all off diagonals
+            off_diagonal = (_np.diagonal(overlap, 1)).tolist()
+            for a in (_np.diagonal(overlap, -1)).tolist():
+                off_diagonal.append(a)
+
+            # check if the off diagonals are 0.03 or larger.
+            too_small = 0
+            for o in off_diagonal:
+                if o < 0.03:
+                    too_small += 1
+            if too_small > 0:
+                message = f"Overlap matrix is bad - {too_small} off-diagonals are less than 0.03."
+                _warnings.warn(f"{message}")
+            else:
+                message = "Overlap matrix is okay."
+
+        return (message)
+
+    def _check_overlap(self):
+        """Check the overlap of an FEP leg. 
+
+           Parameters
+           ----------
+
+           Returns
+           -------
+
+           Message - gives a warning if the overlap is less than 0.03 for any off-diagonal.
+
+
+        """
+
+        # Calculate the overlap for this object.
+        _, overlap = self.analyse()
+
+        # Now call the staticmethod passing in the overlap and the work_dir of the run.
+        return Relative.check_overlap(overlap, estimator=self._estimator)
+
+    @staticmethod
+    def plot(overlap, estimator=None, work_dir=None):
```

Comment:
Just to note that we already have functionality to plot the overlap matrix, which was adapted from @JenkeScheen's code and can be found [here](https://github.com/michellab/BioSimSpace/blob/082f4433a0f77fcecb3696d4b8a3708c9bd32332/python/BioSimSpace/Notebook/_plot.py#L464). I'd rather not have two bits of functionality to do the same thing, so will probably remove the old one.

Reply:
Oh yea! I was trying to keep in consistent so it uses the alchemlyb library and forgot about that.